### DISCUSSION TITLE
RHEL 9.x Support

### DISCUSSION THREAD

**Discussion by ranjujohn** [kubermatic/kubeone#2801] [Q&A] [answered]
Hello,

Does kubeone support RHEL 9.X ? Kubeone [os page](https://docs.kubermatic.com/kubeone/v1.6/architecture/compatibility/os-support-matrix/)  talks about upto RHEL 8.4 only.

**Comment by xmudrii** [comment]
No. We have a tracking issue in the KKP repo to track this: https://github.com/kubermatic/kubermatic/issues/12366

**Comment by ranjujohn** [comment]
@xmudrii What about using RHEL 9 based static nodes for provisioning kubeone cluster?

**Reply by xmudrii** [reply, marked as answer]
~~KubeOne doesn't support RHEL 9 at the moment, so it's mostly like that you can't use it for control plane or static worker nodes. Even if you would be able to use, it might not work properly, and we don't provide support for it at the moment.~~

Edit 2025-12-11: RHEL 9 is supported starting with KubeOne v1.12.

**Reply by ranjujohn** [reply]
@xmudrii Then the only option is to use 8.3 or 8.4(As mentioned on the supported OS page) of RHEL right?

**Reply by xmudrii** [reply]
Yes, exactly. We also accept contributions that would make RHEL 9 work, but please be aware that it would be required to add support both to machine-controller and OSM.

**Reply by ranjujohn** [reply]
Ok. Thank you @xmudrii  for the quick response.

**Comment by tschlaepfer** [comment]
Hi @xmudrii Is there any update on the support for RHEL 9 for Kubeone?

**Reply by kron4eg** [reply]
With the latest and greatest KubeOne v1.12 release now it's supported!

**Comment by kron4eg** [comment]
The latest KubeOne v1.12 release RHEL9 is supported.
